src/mumble/DBus.cpp

```diff
@@ -84,6 +84,34 @@ void MumbleDBus::focus() {
 	g.mw->activateWindow();
 }
 
+/// Change when Mumble transmits voice.
+///
+/// @param mode The new transmit mode (0 = continous, 1 = voice activity, 2 = push-to-talk)
+void MumbleDBus::setTransmitMode(unsigned int mode, const QDBusMessage &msg) {
+	switch (mode) {
+		case 0:
+			g.s.atTransmit = Settings::Continuous;
+			break;
+		case 1:
+			g.s.atTransmit = Settings::VAD;
+			break;
+		case 2:
+			g.s.atTransmit = Settings::PushToTalk;
+			break;
+		default:
+			QDBusConnection::sessionBus().send(msg.createErrorReply(QLatin1String("net.sourceforge.mumble.Error.transmitMode"), QLatin1String("Invalid transmit mode")));
+			return;
+	}
+	g.mw->updateTransmitModeComboBox();
```

Comment:
```suggestion
	QMetaObject::invokeMethod(g.mw, "updateTransmitModeComboBox", Qt::QueuedConnection);
```

We have to be a bit cautious when calling this function as this modifies UI elements (namely the combo box). This is only permitted if done from the main thread (the UI thread). Thus we want to make sure that this function is indeed called from that thread only. Now I'm not sure but I think that the DBus stuff might be running in a separate thread, so this can get an issue.

The suggested code should do the trick, but it is untested. Could you please verify that it actually works? :)

Reply:
It works, but I have to move updateTransmitModeComboBox to `public slots:`. I'm not sure if that has any side-effects, but it seems to work fine.

Reply:
Nah this should be fine :)